docs/notes/2.21.x.md

```diff
@@ -0,0 +1,153 @@
+
+[The `runtime` field](https://www.pantsbuild.org/2.21/reference/targets/python_google_cloud_function#runtime) of the `python_google_cloud_function` target now supports `python312`.
+
+Virtualenvs exported with `pants export ...` have improved in two ways (Note: these apply only when using [`py_resolve_format = "mutable_virtualenv"`](https://www.pantsbuild.org/2.21/reference/goals/export#py_resolve_format), which is the default):
+
+* They include Pants' resolve name in the shell prompt, when activated.
+
+* The [new `py_hermetic_scripts` option](https://www.pantsbuild.org/2.21/reference/goals/export#py_hermetic_scripts) allows disabling the default behaviour of rewriting scripts to be hermetic (ignore system libraries and configuration), to, for example, allow IDEs like PyCharm to inject its debugger, coverage, or other IDE-specific libs when running a script.
+
+Default module mappings were added for more modules: `antlr4-python3-runtime`, `auth0-python`, `biopython`, `cloud-sql-python-connector`, `coolprop`, `databricks-sdk`, `databricks-sql-connetor`, `delta-spark`, `django-activity-stream`, `django-notifications-hq`, `django-oauth-toolkit`, `django-two-factor-auth`, `djangorestframework-jwt`, `dnspython`, `drf-api-tracking`, `faiss-cpu`, `faiss-gpu`, `fonttools`, `grpcio-status`, `grpcio-testing`, `matplotlib`, `matrix-nio`, `netcdf4`, `o365`, `py-healthcheck`, `pycryptodome`, `perfa`, `pynacl`, `pyshp`, `python-sat`, `pywavelets`, `robotraconteur`, `sisl`. The mappings for `opentelemetry` have been restored after a regression in 2.20.0 (NB. this fix was released in 2.20.1 too).
+
+The default version for many subsystems/backends has been updated:
+
+- [`coverage-py`](https://www.pantsbuild.org/2.21/reference/subsystems/coverage-py): 7.2.7
+- [`debugpy`](https://www.pantsbuild.org/2.21/reference/subsystems/debugpy): 1.6.7.post1
+- [`pants.backend.python.lint.autoflake`](https://www.pantsbuild.org/2.21/reference/subsystems/autoflake): 2.1.1
+- [`pants.backend.python.lint.bandit`](https://www.pantsbuild.org/2.21/reference/subsystems/bandit): 1.7.5
+- [`pants.backend.python.lint.black`](https://www.pantsbuild.org/2.21/reference/subsystems/black): 23.3.0
+- [`pants.backend.python.typecheck.mypy`](https://www.pantsbuild.org/2.21/reference/subsystems/mypy): 1.4.1
+- [`pants.backend.python.lint.pyupgrade`](https://www.pantsbuild.org/2.21/reference/subsystems/pyupgrade): 3.3.2
+- [`pants.backend.python.lint.yapf`](https://www.pantsbuild.org/2.21/reference/subsystems/yapf): 0.40.2
+
+The deprecation has expired for the `python_awslambda` target alias: use the [new `python_aws_lambda_function` target](https://www.pantsbuild.org/2.21/reference/targets/python_aws_lambda_function) instead.
```

Comment:
Is this something we should suggest running `update-build-files` for?

Reply:
We could, but this is now a very old deprecation, so I don't think it's worth it.

Reply:
👍🏽